integrations/server_test.go

```diff
@@ -0,0 +1,519 @@
+package integrations
+
+// This was intentionally added under the integrations package and not the integrations test package
+// so that changes in other parts of the code base that may have an effect on these test will not
+// compile until they are fixed.
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/chronograf"
+	"github.com/influxdata/chronograf/bolt"
+	"github.com/influxdata/chronograf/oauth2"
+	"github.com/influxdata/chronograf/server"
+)
+
+func TestServer(t *testing.T) {
+	type fields struct {
+		Organizations []chronograf.Organization
+		Users         []chronograf.User
+		Sources       []chronograf.Source
+		Servers       []chronograf.Server
+		Layouts       []chronograf.Layout
+		Dashboards    []chronograf.Dashboard
+	}
+	type args struct {
+		server    *server.Server
+		method    string
+		path      string
+		payload   interface{} // Expects this to be a json serializable struct
+		principal oauth2.Principal
+	}
+	type wants struct {
+		statusCode  int
+		contentType string
+		body        string
+	}
+
+	tests := []struct {
+		name    string
+		subName string
+		fields  fields
+		args    args
+		wants   wants
+	}{
+		{
+			name:    "GET /users",
+			subName: "User Not Found in the Default Organization",
+			fields:  fields{},
```

Comment:
could we add an explicit empty Users slice to indicate more clearly that there are no users, hence the unauthorized?

Reply:
Good idea.